torch/csrc/jit/script/module.cpp

```diff
@@ -108,11 +108,10 @@ void module_state_to(
     bool non_blocking) {
     // Need to access the `at::Tensor` as a `Variable` here.
     autograd::Variable variable = s.value().toTensor();
-    at::Tensor data = variable.data();
     // Use the data's original device or dtype if not supplied here.
-    auto new_data = data.to(
-        device.value_or(data.device()),
-        dtype.value_or(data.scalar_type()),
+    auto new_data = variable.to(
+        device.value_or(variable.device()),
+        dtype.value_or(variable.scalar_type()),
```

Comment:
does this actually work?  I thought we stopped allow changing the device of a Tensor.

Reply:
I think `variable.to(...)` is returning a new tensor instead of changing the device of `variable` in place. Although the next line `variable.set_data(new_data)` indeed attempts to put the new data (with potentially different device and scalar type) back into `variable`. Since this is also the original semantics of the code, changing the logic here is probably outside of the scope of this PR.